lint-spec.ts

```diff
@@ -113,6 +115,37 @@ function getRelativePath(filepath: string): string {
   return p.relative(process.cwd(), filepath);
 }
 
+async function lintNonHrxTestDir(
+  directory: SpecDirectory,
+  reporter: LintReporter,
+  {fix = false}
+) {
+  if (!directory.isTestDir()) return;
+
+  if (directory instanceof VirtualDirectory) return;
+
+  const options = await directory.options();
+
+  const linterAnnotation =
+    options.getMode('lint-hrx') ?? options.getMode('sass/sass-spec#');
+  // Skip XHR files annotated with :ignore-for: or :todo: containing
+  // `lint-hrx` or `sass/sass-spec#...`
+  if (linterAnnotation !== undefined) return;
+
+  if (!fix) {
+    reporter.reportError(
+      `The test directory "${getRelativePath(
+        directory.path
+      )}" must use the HRX format.`,
+      getReportedInputLocation(directory)
+    );
+  } else {
+    fs.writeFileSync(`${directory.path}.hrx`, await directory.asArchive());
+
+    fs.rmSync(directory.path, {recursive: true});
```

Comment:
This is something I actually wanted to do with my previous PR but for some reason it generated HRX files with wrong encodings 🤔 good thing it worked for you by adding the conditional check! 🙂

Reply:
even better would have been to detect files using BOM or invalid UTF-8 to exclude them automatically. But as we have only a few of them anyway, relying on the ignore rule is simpler for now.